### PR TITLE
build: scripted-diff: drop config/ subdir for bitcoin-config.h

### PR DESCRIPTION
**Pull request by theStack**
This PR is a follow-up to #30856, as suggested in comment https://github.com/bitcoin/bitcoin/pull/30856#issuecomment-2356804690. With the scripted diff, review should be fairly trivial, but it could still be seen as controversial due to the large number of files (78 in total) being touched.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK, given the early stage of the current release cycle and the presence of only a single conflicting PR.

**Review by theuni — COMMENTED** [review]
Concept ACK, and scripted-diff lgtm ACK.

Bikeshed: How about `bitcoin-build-config.h` for symmetry with `bitcoin-build-info.h`?

File: test/lint/test_runner/src/main.rs [hunk 1]
```diff
@@ -429,7 +429,7 @@ fn lint_includes_build_config() -> LintResult {
                 ])
                 .args(get_pathspecs_exclude_subtrees())
                 .args([
-                    // These are exceptions which don't use bitcoin-config.h, rather the Makefile.am adds
+                    // These are exceptions which don't use bitcoin-build-config.h, rather the Makefile.am adds
```
Comment (hebasto) [inline test/lint/test_runner/src/main.rs:432]: Not directly related to this PR change, but mentioning "Makefile.am" is outdated.

**Review by hebasto — APPROVED** [review]
ACK 09b0161c4a2502564f50fa2b3ce9b19d8fc40d8b.

**Review by theuni — APPROVED** [review]
utACK 09b0161c4a2502564f50fa2b3ce9b19d8fc40d8b

**Review by sedited — APPROVED** [review]
ACK 09b0161c4a2502564f50fa2b3ce9b19d8fc40d8b

**Comment by fanquake** [comment]
Guix Build
```bash
a78d6d030a78972bf2ebab6d3599a3ce8b325e5422e192ab97d65966f3ee0ae4  guix-build-09b0161c4a25/output/aarch64-linux-gnu/SHA256SUMS.part
a3ff2330715466bef18e536f5ef4c6c00c9a359e11f896b3afade7364b4e40eb  guix-build-09b0161c4a25/output/aarch64-linux-gnu/bitcoin-09b0161c4a25-aarch64-linux-gnu-debug.tar.gz
dcedb6cea8f7799d34d64b9a73deda40452887898a44e6b5a2048b7ebf304924  guix-build-09b0161c4a25/output/aarch64-linux-gnu/bitcoin-09b0161c4a25-aarch64-linux-gnu.tar.gz
aa2b1c6e8fc747872bdb1c2cbbf5622c96736781a98cc5e6209460d8c0454891  guix-build-09b0161c4a25/output/arm-linux-gnueabihf/SHA256SUMS.part
1964e1c6a54a7dfe462a2f28bf7e4d05e66baf32b1193d8b3314989e5de129de  guix-build-09b0161c4a25/output/arm-linux-gnueabihf/bitcoin-09b0161c4a25-arm-linux-gnueabihf-debug.tar.gz
d2922f38049d5907ecc5120305673e0f4e3cd23b6a129bc5fd29b420e0f7e5aa  guix-build-09b0161c4a25/output/arm-linux-gnueabihf/bitcoin-09b0161c4a25-arm-linux-gnueabihf.tar.gz
05d473e74b2707834f0147f3e42a8e328f40fe8ba0a2df31d3d3e36656a07296  guix-build-09b0161c4a25/output/arm64-apple-darwin/SHA256SUMS.part
ddeb1a9ef5ea95481aedf6eb6f3fda7829711cac5f4c0e36f8e977fc3391cda4  guix-build-09b0161c4a25/output/arm64-apple-darwin/bitcoin-09b0161c4a25-arm64-apple-darwin-unsigned.tar.gz
e502cee115489c03e0c38f76fde5dfeba72266046eb35bc9b1a0fbe72670fd51  guix-build-09b0161c4a25/output/arm64-apple-darwin/bitcoin-09b0161c4a25-arm64-apple-darwin-unsigned.zip
66ab20ebf87a11b41149844ae1fc0c148d60059027917ee835547d853bb58b53  guix-build-09b0161c4a25/output/arm64-apple-darwin/bitcoin-09b0161c4a25-arm64-apple-darwin.tar.gz
b7dad4ad285c4eecb11dab43e01435da206aeb3dbbd264a51575d0300b5bec6d  guix-build-09b0161c4a25/output/dist-archive/bitcoin-09b0161c4a25.tar.gz
ad7c3270dfe8955b9a518b081d6c4c45322f8627742be111d95c1c7891aa34de  guix-build-09b0161c4a25/output/powerpc64-linux-gnu/SHA256SUMS.part
434208f50ae66ce5eb9df81337586f4e69f39228fd620a8570390506eccacb45  guix-build-09b0161c4a25/output/powerpc64-linux-gnu/bitcoin-09b0161c4a25-powerpc64-linux-gnu-debug.tar.gz
45fba4d2324a8bbe543e062802652aa2518475ae92a62bb053e94506231dc268  guix-build-09b0161c4a25/output/powerpc64-linux-gnu/bitcoin-09b0161c4a25-powerpc64-linux-gnu.tar.gz
d09cddd15788c640c018c82d066386ed41b3a79f22b311ef5fab210afe06ca8a  guix-build-09b0161c4a25/output/riscv64-linux-gnu/SHA256SUMS.part
3f68083240793dc897c25a303748af989e87b99e3bae7cf6370339a42b2ff4b0  guix-build-09b0161c4a25/output/riscv64-linux-gnu/bitcoin-09b0161c4a25-riscv64-linux-gnu-debug.tar.gz
c1c66255621ffe348113c7b57ff0432cd420a10dd008c3dfcc6c0caf787b7618  guix-build-09b0161c4a25/output/riscv64-linux-gnu/bitcoin-09b0161c4a25-riscv64-linux-gnu.tar.gz
99988b30ae435a6e5155631f683f1934ee5f3e3e2d7655fb4034c9bea2537fd5  guix-build-09b0161c4a25/output/x86_64-apple-darwin/SHA256SUMS.part
5ce76d80d52e5b9c44599b7189276f81beda58137f0e232d9f22116e6a44289f  guix-build-09b0161c4a25/output/x86_64-apple-darwin/bitcoin-09b0161c4a25-x86_64-apple-darwin-unsigned.tar.gz
8c195e6ef2ef8a6073381d08e76746e9109b641041e7c7efd7f9843c88609160  guix-build-09b0161c4a25/output/x86_64-apple-darwin/bitcoin-09b0161c4a25-x86_64-apple-darwin-unsigned.zip
851914c98027dbd4219d94b81d337e34f3509ae986a3dd2499f6844615e4f284  guix-build-09b0161c4a25/output/x86_64-apple-darwin/bitcoin-09b0161c4a25-x86_64-apple-darwin.tar.gz
268ddadba8dd0e37e344e125ca2a37e6ee9945b5ab707968f65997031ce09990  guix-build-09b0161c4a25/output/x86_64-linux-gnu/SHA256SUMS.part
1e90e024edf115c25795d73ab1fd328bc4e65f32240ca3ac3a1f0b3299bb7960  guix-build-09b0161c4a25/output/x86_64-linux-gnu/bitcoin-09b0161c4a25-x86_64-linux-gnu-debug.tar.gz
176429859b3f7cc295b7361a4692fe5f5927357f5762982c1cbb2bc412447cb4  guix-build-09b0161c4a25/output/x86_64-linux-gnu/bitcoin-09b0161c4a25-x86_64-linux-gnu.tar.gz
786f56a48dda53fa3a6387114d131018e36e1955fc771ee33e61f7c4c85ff480  guix-build-09b0161c4a25/output/x86_64-w64-mingw32/SHA256SUMS.part
04eb6e213da49d1232ab6312e42c5b2f901dcad33f72d3cbc77e2310295cddbf  guix-build-09b0161c4a25/output/x86_64-w64-mingw32/bitcoin-09b0161c4a25-win64-debug.zip
bc2a8d194cd12afd801f8bde5f0d384006b66806705b8fa74f474a8777a1cdc9  guix-build-09b0161c4a25/output/x86_64-w64-mingw32/bitcoin-09b0161c4a25-win64-setup-unsigned.exe
af36eedcf415b94e8e6323fd897710d0ac51efdc6eddb580a85d38992abbced3  guix-build-09b0161c4a25/output/x86_64-w64-mingw32/bitcoin-09b0161c4a25-win64-unsigned.tar.gz
930a85398e43ee9e9af1d1cdccb1f1c472114167995cc15a8be1e3961cfe01e4  guix-build-09b0161c4a25/output/x86_64-w64-mingw32/bitcoin-09b0161c4a25-win64.zip
```

**Comment by fanquake** [comment]
```bash
bitcoin/src/common/netif.cpp:5:10: fatal error: 'config/bitcoin-config.h' file not found
    5 | #include <config/bitcoin-config.h> // IWYU pragma: keep
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by fanquake** [comment]
Guix Build:
```bash
9cf65a99fa046a1a3acbee14d6dec658b9fe7f4e55a4b6f50d063ebb4029e3eb  guix-build-882f736d0a60/output/aarch64-linux-gnu/SHA256SUMS.part
46780332298b988bc9b4c2903d077f49978dc92250cf5221557ce4459383b18e  guix-build-882f736d0a60/output/aarch64-linux-gnu/bitcoin-882f736d0a60-aarch64-linux-gnu-debug.tar.gz
72362b6f0394f4b8b1e69e4520c892d5b3e9f3d86788c8f0ee59f2b0d9e51717  guix-build-882f736d0a60/output/aarch64-linux-gnu/bitcoin-882f736d0a60-aarch64-linux-gnu.tar.gz
c4b5294efed908afb3458d1af340c9d2f77be84e0a9db93f626bad4590fe3fbe  guix-build-882f736d0a60/output/arm-linux-gnueabihf/SHA256SUMS.part
e9d201fc4f74fa77b038ca165858efde284f82e8e004301290c3c0d61875f1c1  guix-build-882f736d0a60/output/arm-linux-gnueabihf/bitcoin-882f736d0a60-arm-linux-gnueabihf-debug.tar.gz
8cd166df49767e948a2f827e148ab16f57c6910a35fe6b39ff3373e9c46cb936  guix-build-882f736d0a60/output/arm-linux-gnueabihf/bitcoin-882f736d0a60-arm-linux-gnueabihf.tar.gz
f9ac382c47e7e17b4ed6a0e0666c894967dcda4289a139e9d41e88d78b4f8ee6  guix-build-882f736d0a60/output/arm64-apple-darwin/SHA256SUMS.part
cdfd080d9b6a2e8f9f83a6f51c8f7db0e77d97a3f3ed4ecd1aab6293805db664  guix-build-882f736d0a60/output/arm64-apple-darwin/bitcoin-882f736d0a60-arm64-apple-darwin-unsigned.tar.gz
80f1a45978a17e72042e9e9cd02972ceb5eabfaa04b4a1dbbb83a8710cff64a0  guix-build-882f736d0a60/output/arm64-apple-darwin/bitcoin-882f736d0a60-arm64-apple-darwin-unsigned.zip
7f11308b9981b417be87029eb0a04aad36335874f0c1e7e5101a2161f3d62d7c  guix-build-882f736d0a60/output/arm64-apple-darwin/bitcoin-882f736d0a60-arm64-apple-darwin.tar.gz
32ee2152bf29b81e8c40dbd7c5821b646c498447f63d51ca9fbe414dc485bec0  guix-build-882f736d0a60/output/dist-archive/bitcoin-882f736d0a60.tar.gz
530692b5ce8dd83150e838a6e2e4fa042662ee688db79d874e50110de23987f2  guix-build-882f736d0a60/output/powerpc64-linux-gnu/SHA256SUMS.part
06810c70111b01ef998031a4fe1b5d661a7b89287f2f36e64e9991b38aac99e8  guix-build-882f736d0a60/output/powerpc64-linux-gnu/bitcoin-882f736d0a60-powerpc64-linux-gnu-debug.tar.gz
fbefb433c08ea43f0ba3095307ed0981bd47e2954799eca23eb9f7e7bb78cfab  guix-build-882f736d0a60/output/powerpc64-linux-gnu/bitcoin-882f736d0a60-powerpc64-linux-gnu.tar.gz
e605fcdfe071b4b504a9033c175f3d9651dffb3ef959fcc0f34082dd2815cc3d  guix-build-882f736d0a60/output/riscv64-linux-gnu/SHA256SUMS.part
1104a13281c11351b0672a8fee0384d9fd9b2becfe1adc168eda07c807edd5a8  guix-build-882f736d0a60/output/riscv64-linux-gnu/bitcoin-882f736d0a60-riscv64-linux-gnu-debug.tar.gz
ef1c20cf3b798b9cd6d05a7ade98f4c3b8f0e7afb9a139e5975d204f8aa58985  guix-build-882f736d0a60/output/riscv64-linux-gnu/bitcoin-882f736d0a60-riscv64-linux-gnu.tar.gz
9eeb7155477ec4db1abd308a9a036945c442f11f7ed6aa52ea9a1b825babe57b  guix-build-882f736d0a60/output/x86_64-apple-darwin/SHA256SUMS.part
c78b42f3aa52bb424e7e8aad650343e582a36ab00b5dc84c3e922c99470be7f7  guix-build-882f736d0a60/output/x86_64-apple-darwin/bitcoin-882f736d0a60-x86_64-apple-darwin-unsigned.tar.gz
04f78c2e0674e9ff6a1358b4322b045489940925c37c504503c8a765b19a1603  guix-build-882f736d0a60/output/x86_64-apple-darwin/bitcoin-882f736d0a60-x86_64-apple-darwin-unsigned.zip
66d5a64c486eefd899e37a29a1aabc53fb8ba33f46bb5cb83399db8135470ba9  guix-build-882f736d0a60/output/x86_64-apple-darwin/bitcoin-882f736d0a60-x86_64-apple-darwin.tar.gz
f4d169df64d85d835b0baa46d12b2992dbf1a1bfdca5a136c94f7794b11df28a  guix-build-882f736d0a60/output/x86_64-linux-gnu/SHA256SUMS.part
60c34fcd5b5fa08afcc3a7ee65a8d820820a1af9f42f731ec8422e6db458d616  guix-build-882f736d0a60/output/x86_64-linux-gnu/bitcoin-882f736d0a60-x86_64-linux-gnu-debug.tar.gz
31135e0adbeb59f0abe78351a7c260aaf02d3206a40b26dfda4b20d8b6919627  guix-build-882f736d0a60/output/x86_64-linux-gnu/bitcoin-882f736d0a60-x86_64-linux-gnu.tar.gz
c8e17374b9e6395215e1b2fba46087c6e1b662f5cae3cd6a532cebc0810ffcf9  guix-build-882f736d0a60/output/x86_64-w64-mingw32/SHA256SUMS.part
ef9b885e8a28f02bf5fd904aaac1a6a855abe9a1c460f879d510e34e759bf18f  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64-debug.zip
f7f5d508293ab7981b32717285e34206acbf7d2aaac01460d6183f9ff767c877  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64-setup-unsigned.exe
af63e56853250b509ab4cdc63b24b195d176d4a4e943ec8bb04b58ca559bf750  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64-unsigned.tar.gz
0b4810fea2c99c2c23bc2edea341b06df19070e9b4e31d47aa8ddf22ca98018f  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64.zip
```

**Review by fanquake — APPROVED** [review]
ACK 882f736d0a607976ee5e3a6cbcb5385524bc72c6

**Comment by TheCharlatan** [comment]
Post-merge ACK 882f736d0a60

Guix build (aarch64):
```
9cf65a99fa046a1a3acbee14d6dec658b9fe7f4e55a4b6f50d063ebb4029e3eb  guix-build-882f736d0a60/output/aarch64-linux-gnu/SHA256SUMS.part
46780332298b988bc9b4c2903d077f49978dc92250cf5221557ce4459383b18e  guix-build-882f736d0a60/output/aarch64-linux-gnu/bitcoin-882f736d0a60-aarch64-linux-gnu-debug.tar.gz
72362b6f0394f4b8b1e69e4520c892d5b3e9f3d86788c8f0ee59f2b0d9e51717  guix-build-882f736d0a60/output/aarch64-linux-gnu/bitcoin-882f736d0a60-aarch64-linux-gnu.tar.gz
c4b5294efed908afb3458d1af340c9d2f77be84e0a9db93f626bad4590fe3fbe  guix-build-882f736d0a60/output/arm-linux-gnueabihf/SHA256SUMS.part
e9d201fc4f74fa77b038ca165858efde284f82e8e004301290c3c0d61875f1c1  guix-build-882f736d0a60/output/arm-linux-gnueabihf/bitcoin-882f736d0a60-arm-linux-gnueabihf-debug.tar.gz
8cd166df49767e948a2f827e148ab16f57c6910a35fe6b39ff3373e9c46cb936  guix-build-882f736d0a60/output/arm-linux-gnueabihf/bitcoin-882f736d0a60-arm-linux-gnueabihf.tar.gz
f9ac382c47e7e17b4ed6a0e0666c894967dcda4289a139e9d41e88d78b4f8ee6  guix-build-882f736d0a60/output/arm64-apple-darwin/SHA256SUMS.part
cdfd080d9b6a2e8f9f83a6f51c8f7db0e77d97a3f3ed4ecd1aab6293805db664  guix-build-882f736d0a60/output/arm64-apple-darwin/bitcoin-882f736d0a60-arm64-apple-darwin-unsigned.tar.gz
80f1a45978a17e72042e9e9cd02972ceb5eabfaa04b4a1dbbb83a8710cff64a0  guix-build-882f736d0a60/output/arm64-apple-darwin/bitcoin-882f736d0a60-arm64-apple-darwin-unsigned.zip
7f11308b9981b417be87029eb0a04aad36335874f0c1e7e5101a2161f3d62d7c  guix-build-882f736d0a60/output/arm64-apple-darwin/bitcoin-882f736d0a60-arm64-apple-darwin.tar.gz
32ee2152bf29b81e8c40dbd7c5821b646c498447f63d51ca9fbe414dc485bec0  guix-build-882f736d0a60/output/dist-archive/bitcoin-882f736d0a60.tar.gz
530692b5ce8dd83150e838a6e2e4fa042662ee688db79d874e50110de23987f2  guix-build-882f736d0a60/output/powerpc64-linux-gnu/SHA256SUMS.part
06810c70111b01ef998031a4fe1b5d661a7b89287f2f36e64e9991b38aac99e8  guix-build-882f736d0a60/output/powerpc64-linux-gnu/bitcoin-882f736d0a60-powerpc64-linux-gnu-debug.tar.gz
fbefb433c08ea43f0ba3095307ed0981bd47e2954799eca23eb9f7e7bb78cfab  guix-build-882f736d0a60/output/powerpc64-linux-gnu/bitcoin-882f736d0a60-powerpc64-linux-gnu.tar.gz
e605fcdfe071b4b504a9033c175f3d9651dffb3ef959fcc0f34082dd2815cc3d  guix-build-882f736d0a60/output/riscv64-linux-gnu/SHA256SUMS.part
1104a13281c11351b0672a8fee0384d9fd9b2becfe1adc168eda07c807edd5a8  guix-build-882f736d0a60/output/riscv64-linux-gnu/bitcoin-882f736d0a60-riscv64-linux-gnu-debug.tar.gz
ef1c20cf3b798b9cd6d05a7ade98f4c3b8f0e7afb9a139e5975d204f8aa58985  guix-build-882f736d0a60/output/riscv64-linux-gnu/bitcoin-882f736d0a60-riscv64-linux-gnu.tar.gz
9eeb7155477ec4db1abd308a9a036945c442f11f7ed6aa52ea9a1b825babe57b  guix-build-882f736d0a60/output/x86_64-apple-darwin/SHA256SUMS.part
c78b42f3aa52bb424e7e8aad650343e582a36ab00b5dc84c3e922c99470be7f7  guix-build-882f736d0a60/output/x86_64-apple-darwin/bitcoin-882f736d0a60-x86_64-apple-darwin-unsigned.tar.gz
04f78c2e0674e9ff6a1358b4322b045489940925c37c504503c8a765b19a1603  guix-build-882f736d0a60/output/x86_64-apple-darwin/bitcoin-882f736d0a60-x86_64-apple-darwin-unsigned.zip
66d5a64c486eefd899e37a29a1aabc53fb8ba33f46bb5cb83399db8135470ba9  guix-build-882f736d0a60/output/x86_64-apple-darwin/bitcoin-882f736d0a60-x86_64-apple-darwin.tar.gz
f4d169df64d85d835b0baa46d12b2992dbf1a1bfdca5a136c94f7794b11df28a  guix-build-882f736d0a60/output/x86_64-linux-gnu/SHA256SUMS.part
60c34fcd5b5fa08afcc3a7ee65a8d820820a1af9f42f731ec8422e6db458d616  guix-build-882f736d0a60/output/x86_64-linux-gnu/bitcoin-882f736d0a60-x86_64-linux-gnu-debug.tar.gz
31135e0adbeb59f0abe78351a7c260aaf02d3206a40b26dfda4b20d8b6919627  guix-build-882f736d0a60/output/x86_64-linux-gnu/bitcoin-882f736d0a60-x86_64-linux-gnu.tar.gz
c8e17374b9e6395215e1b2fba46087c6e1b662f5cae3cd6a532cebc0810ffcf9  guix-build-882f736d0a60/output/x86_64-w64-mingw32/SHA256SUMS.part
ef9b885e8a28f02bf5fd904aaac1a6a855abe9a1c460f879d510e34e759bf18f  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64-debug.zip
f7f5d508293ab7981b32717285e34206acbf7d2aaac01460d6183f9ff767c877  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64-setup-unsigned.exe
af63e56853250b509ab4cdc63b24b195d176d4a4e943ec8bb04b58ca559bf750  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64-unsigned.tar.gz
0b4810fea2c99c2c23bc2edea341b06df19070e9b4e31d47aa8ddf22ca98018f  guix-build-882f736d0a60/output/x86_64-w64-mingw32/bitcoin-882f736d0a60-win64.zip
```

**Review by hebasto — COMMENTED** [review]
Post-merge ACK 882f736d0a607976ee5e3a6cbcb5385524bc72c6.

**Pull request merged**